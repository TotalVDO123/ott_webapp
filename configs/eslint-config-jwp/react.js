module.exports = {
  extends: [
    // Extend the base config
    './typescript',

    // Use recommended React rules
    'plugin:react/recommended',
  ],

  rules: {
    // Not needed in React 17
    'react/react-in-jsx-scope': 'off',
  },

  overrides: [
    {
      files: ['*.jsx', '*.tsx', '*.ts'],
      plugins: [
        // Enable linting React code
        'react',
        'react-hooks',
      ],
      rules: {
        // Help with Hooks syntax
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'error',

        // Handled by Typescript
        'react/prop-types': 'off',

        // This rule causes too many false positives, e.g. with default exports or child render function
        'react/display-name': 'off',
      },
    },
  ],

  settings: {
    react: {
      pragma: 'React',
      version: '17',
    },
  },
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
};
