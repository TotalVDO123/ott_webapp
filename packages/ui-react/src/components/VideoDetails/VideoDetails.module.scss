@use '@jwp/ott-ui-react/src/styles/variables';
@use '@jwp/ott-ui-react/src/styles/theme';
@use '@jwp/ott-ui-react/src/styles/mixins/responsive';
@use '@jwp/ott-ui-react/src/styles/mixins/typography';

.videoDetails {
  font-family: var(--body-font-family), sans-serif;
}

.metaContainer {
  display: flex;
  flex-direction: column;
}

.primaryMetadata {
  display: flex;
  align-items: center;
  margin-bottom: 8px;
  font-size: 16px;
  line-height: variables.$base-line-height;
  letter-spacing: 0.15px;

  @include responsive.mobile-only() {
    font-size: 14px;
  }
}

.secondaryMetadata {
  margin-top: 24px;
  margin-bottom: 8px;
  font-size: 20px;
  line-height: variables.$base-line-height;
  letter-spacing: 0.5px;

  @include responsive.mobile-only() {
    order: 1;
    margin: 4px 0;
    font-size: 18px;
  }
}

.description {
  margin-bottom: 24px;
}

.buttonBar {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
  gap: calc(variables.$base-spacing / 2);
  width: 100%;

  > button {
    flex: 1;
    justify-content: center;

    &:first-child {
      flex-basis: 100%;
    }
  }

  @include responsive.tablet-and-larger() {
    max-width: 400px;
  }
}

.playIcon {
  position: absolute;
  top: calc(50% - 60px);
  left: calc(50% - 35px);
  padding: 25px 30px;
  font-size: xx-large;
  border: 3px solid white;
  border-radius: 100%;
}
