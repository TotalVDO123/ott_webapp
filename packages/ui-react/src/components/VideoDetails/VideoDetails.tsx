import React, { type PropsWithChildren } from 'react';
import { testId } from '@jwp/ott-common/src/utils/common';

import Hero from '../Hero/Hero';
import HeroTitle from '../Hero/HeroTitle';
import HeroDescription from '../Hero/HeroDescription';

import styles from './VideoDetails.module.scss';

type Props = PropsWithChildren<{
  title: string;
  description: string;
  primaryMetadata: React.ReactNode;
  secondaryMetadata?: React.ReactNode;
  image?: string;
  startWatchingButton: React.ReactNode;
  shareButton: React.ReactNode;
  favoriteButton?: React.ReactNode;
  trailerButton?: React.ReactNode;
}>;

const VideoDetails = ({
  title,
  description,
  primaryMetadata,
  secondaryMetadata,
  image,
  startWatchingButton,
  shareButton,
  favoriteButton,
  trailerButton,
  children,
}: Props) => {
  return (
    <div data-testid={testId('cinema-layout')}>
      <header className={styles.videoDetails} data-testid={testId('video-details')} id="video-details">
        <Hero image={image}>
          <HeroTitle title={title} />
          <div className={styles.metaContainer}>
            <div className={styles.primaryMetadata}>{primaryMetadata}</div>
            {secondaryMetadata && <div className={styles.secondaryMetadata}>{secondaryMetadata}</div>}
          </div>
          <HeroDescription className={styles.description} description={description} />
          <div className={styles.buttonBar}>
            {startWatchingButton}
            {trailerButton}
            {favoriteButton}
            {shareButton}
          </div>
        </Hero>
      </header>
      {children}
    </div>
  );
};

export default VideoDetails;
