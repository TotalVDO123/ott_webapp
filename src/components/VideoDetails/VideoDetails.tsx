import React from 'react';
import classNames from 'classnames';
import { useTranslation } from 'react-i18next';

import styles from './VideoDetails.module.scss';

import CollapsibleText from '#src/components/CollapsibleText/CollapsibleText';
import Button from '#src/components/Button/Button';
import useBreakpoint, { Breakpoint } from '#src/hooks/useBreakpoint';
import Favorite from '#src/icons/Favorite';
import PlayTrailer from '#src/icons/PlayTrailer';
import FavoriteBorder from '#src/icons/FavoriteBorder';

type PosterMode = 'fading' | 'normal';

type Props = {
  title: string;
  description: string;
  primaryMetadata: React.ReactNode;
  secondaryMetadata?: React.ReactNode;
  isFavorite: boolean;
  isFavoritesEnabled: boolean;
  onFavoriteButtonClick: () => void;
  poster?: string;
  posterMode: PosterMode;
  hasTrailer: boolean;
  onTrailerClick: () => void;
  playTrailer: boolean;
  startWatchingButton: React.ReactNode;
  shareButton: React.ReactNode;
};

const VideoDetails: React.FC<Props> = ({
  title,
  description,
  primaryMetadata,
  secondaryMetadata,
  poster,
  posterMode,
  isFavorite,
  isFavoritesEnabled,
  onFavoriteButtonClick,
  children,
  hasTrailer,
  playTrailer,
  onTrailerClick,
  startWatchingButton,
  shareButton,
}) => {
  const breakpoint: Breakpoint = useBreakpoint();
  const { t } = useTranslation(['video', 'common']);

  const isMobile = breakpoint === Breakpoint.xs;

  return (
    <div className={styles.video}>
      <div
        className={classNames(styles.main, styles.mainPadding, {
          [styles.posterNormal]: poster === 'normal',
        })}
      >
        <div className={styles.info}>
          <h2 className={styles.title}>{title}</h2>
          <div className={styles.metaContainer}>
            <div className={styles.primaryMetadata}>{primaryMetadata}</div>
            {secondaryMetadata && <div className={styles.secondaryMetadata}>{secondaryMetadata}</div>}
          </div>
          <CollapsibleText text={description} className={styles.description} maxHeight={isMobile ? 60 : 'none'} />

          <div className={styles.buttonBar}>
            {startWatchingButton}
            {hasTrailer && (
              <Button
                className={styles.bigButton}
                label={t('video:trailer')}
                aria-label={t('video:watch_trailer')}
                startIcon={<PlayTrailer />}
                onClick={onTrailerClick}
                active={playTrailer}
                fullWidth={breakpoint < Breakpoint.md}
              />
            )}
            {isFavoritesEnabled && (
              <Button
                label={t('video:favorite')}
                aria-label={isFavorite ? t('video:remove_from_favorites') : t('video:add_to_favorites')}
                startIcon={isFavorite ? <Favorite /> : <FavoriteBorder />}
                onClick={onFavoriteButtonClick}
                color={isFavorite ? 'primary' : 'default'}
                fullWidth={breakpoint < Breakpoint.md}
              />
            )}
            {shareButton}
          </div>
        </div>
        <div className={classNames(styles.poster, styles[posterMode])} style={{ backgroundImage: `url('${poster}')` }} />
      </div>
      {!!children && <div className={classNames(styles.related, styles.mainPadding)}>{children}</div>}
    </div>
  );
};

export default VideoDetails;
