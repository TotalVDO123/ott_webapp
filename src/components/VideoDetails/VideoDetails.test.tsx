import React from 'react';
import { render } from '@testing-library/react';

import VideoDetails from './VideoDetails';

describe('<VideoDetails>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(
      <VideoDetails
        title="Test video"
        description="Video description"
        videoMeta="Video meta"
        poster="posterimage"
        posterMode="fading"
        hasTrailer={true}
        isFavorite={false}
        isFavoritesEnabled={true}
        onFavoriteButtonClick={vi.fn()}
        playTrailer={false}
        onTrailerClick={vi.fn()}
        onTrailerClose={vi.fn()}
        startWatchingButton={<button>Start watching</button>}
        shareButton={<button>share</button>}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
