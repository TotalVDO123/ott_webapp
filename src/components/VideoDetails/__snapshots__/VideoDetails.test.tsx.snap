// Vitest Snapshot v1

exports[`<VideoDetails> > renders and matches snapshot 1`] = `
<div>
  <div
    class="video"
  >
    <div
      class="main mainPadding"
    >
      <div
        class="info"
      >
        <h2
          class="title"
        >
          Test video
        </h2>
        <div
          class="metaContainer"
        >
          <div
            class="primaryMetadata"
          >
            Primary metadata string
          </div>
          <div
            class="secondaryMetadata"
          >
            <strong>
              Secondary metadata string
            </strong>
          </div>
        </div>
        <div
          class="collapsibleText"
        >
          <div
            class="textContainer description"
            style="max-height: none;"
          >
            Video description
          </div>
        </div>
        <div
          class="buttonBar"
        >
          <button>
            Start watching
          </button>
          <button
            aria-label="video:watch_trailer"
            class="button default outlined"
          >
            <div
              class="startIcon"
            >
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M19,19H5V5H19M19,3H5A2,2 0 0,0 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3M10,8V16L15,12L10,8Z"
                  />
                </g>
              </svg>
            </div>
            <span>
              video:trailer
            </span>
          </button>
          <button
            aria-label="video:add_to_favorites"
            class="button default outlined"
          >
            <div
              class="startIcon"
            >
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12.1,18.55L12,18.65L11.89,18.55C7.14,14.24 4,11.39 4,8.5C4,6.5 5.5,5 7.5,5C9.04,5 10.54,6 11.07,7.36H12.93C13.46,6 14.96,5 16.5,5C18.5,5 20,6.5 20,8.5C20,11.39 16.86,14.24 12.1,18.55M16.5,3C14.76,3 13.09,3.81 12,5.08C10.91,3.81 9.24,3 7.5,3C4.42,3 2,5.41 2,8.5C2,12.27 5.4,15.36 10.55,20.03L12,21.35L13.45,20.03C18.6,15.36 22,12.27 22,8.5C22,5.41 19.58,3 16.5,3Z"
                  />
                </g>
              </svg>
            </div>
            <span>
              video:favorite
            </span>
          </button>
          <button>
            share
          </button>
        </div>
      </div>
      <div
        class="poster fading"
        style="background-image: url(posterimage);"
      />
    </div>
  </div>
</div>
`;
